pkg/route/registry/route/strategy.go

```diff
@@ -67,7 +69,36 @@ func (s routeStrategy) PrepareForUpdate(ctx apirequest.Context, obj, old runtime
 // allocateHost allocates a host name ONLY if the route doesn't specify a subdomain wildcard policy and
 // the host name on the route is empty and an allocator is configured.
 // It must first allocate the shard and may return an error if shard allocation fails.
-func (s routeStrategy) allocateHost(route *api.Route) error {
+func (s routeStrategy) allocateHost(ctx apirequest.Context, route *api.Route) field.ErrorList {
+	hostSet := len(route.Spec.Host) > 0
+	certSet := route.Spec.TLS != nil && (len(route.Spec.TLS.CACertificate) > 0 || len(route.Spec.TLS.Certificate) > 0 || len(route.Spec.TLS.DestinationCACertificate) > 0 || len(route.Spec.TLS.Key) > 0)
+	if hostSet || certSet {
+		user, ok := apirequest.UserFrom(ctx)
+		if !ok {
+			return field.ErrorList{field.InternalError(field.NewPath("spec", "host"), fmt.Errorf("unable to verify host field can be set"))}
+		}
+		res, err := s.sarClient.CreateSubjectAccessReview(
+			ctx,
+			&authorizationapi.SubjectAccessReview{
+				User: user.GetName(),
```

Comment:
nm, @deads2k picked it up in https://github.com/openshift/origin/pull/14285/commits/26cd8fd2090c31006a42bb58aa3664638f4fd2f4#diff-13cdc93be4a264426a1f7c00583e996dR82